opentelemetry-sdk/src/propagation/baggage.rs

```diff
@@ -1,45 +1,3 @@
-//! # OpenTelemetry Baggage API
```

Comment:
Removed this as it's duplicate with the `BaggatePropagator` doc

Reply:
Thanks. 

Separately, https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry/src/baggage.rs#L393-L399 has some extra things like `BaggageMetadata` that are not in spec.. Need to review them to ensure its needed.

Reply:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md#propagators-distribution 
Also, the propagators should be part of the API itself.. 

Not a blocker for this PR, will track separately.

Reply:
Is there a ticket to track?

Reply:
No, haven;t create an issue to track yet.

Reply:
> Also can't we reference it here without needing to duplicate?

We don' really expose this module(we export the `BaggagePropagator` directly) so users won't see it.